File: ios/Tehilim/Core/Models/DailyRules.swift
import Foundation

enum DailyMode: String, Codable, CaseIterable, Identifiable {
    case monthly, weekly, custom
    var id: String { rawValue }

    var label: String {
        switch self {
        case .monthly: return String(localized: "Cycle mensuel")
        case .weekly:  return String(localized: "Jour de la semaine")
        case .custom:  return String(localized: "Personnalisé")
        }
    }
}

enum Weekday: String, Codable, CaseIterable {
    case sunday, monday, tuesday, wednesday, thursday, friday, saturday

    static func from(_ d: Date, calendar: Calendar = .current) -> Weekday {
        let comp = calendar.component(.weekday, from: d) // 1 = Sunday
        return Weekday.allCases[comp - 1]
    }
}

struct DailyRules: Codable {
    struct Modes: Codable {
        let monthly: Monthly
        let weekly: Weekly
        let custom: Custom?
    }
    struct Monthly: Codable {
        let days: [String: [Int]]          // "1".."29"/"30"
    }
    struct Weekly: Codable {
        let days: [String: [Int]]          // "sunday"... "saturday"
    }
    struct Custom: Codable {
        let enabled: Bool
    }

    let version: String
    let modes: Modes

    static let empty = DailyRules(
        version: "0.0.0",
        modes: .init(
            monthly: .init(days: [:]),
            weekly: .init(days: [:]),
            custom: .init(enabled: false)
        )
    )
}

File: android/app/src/main/java/com/david/tehilim/core/model/DailyRules.kt
package com.david.tehilim.core.model

import kotlinx.serialization.Serializable

/**
 * Règles de répartition des 150 Tehilim sur les jours.
 * Structure réelle du JSON :
 *   modes.monthly.days.{1..30} = List<Int>
 *   modes.weekly.days.{sunday..saturday} = List<Int>
 */
@Serializable
data class DailyRules(
    val modes: Modes = Modes()
) {
    @Serializable
    data class Modes(
        val monthly: ModeBlock = ModeBlock(),
        val weekly: ModeBlock = ModeBlock(),
        val custom: ModeBlock = ModeBlock()
    )

    @Serializable
    data class ModeBlock(
        val description: String = "",
        val days: Map<String, List<Int>> = emptyMap()
    )

    /** Tehilim pour un jour donné, en mode mensuel (clé = jour numérique 1..30 en String). */
    fun monthly(day: Int): List<Int> = modes.monthly.days[day.toString()] ?: emptyList()

    /** Tehilim pour un jour donné, en mode hebdomadaire (clé = nom anglais minuscule). */
    fun weekly(dayKey: String): List<Int> = modes.weekly.days[dayKey] ?: emptyList()

    companion object {
        val empty = DailyRules()
    }
}

/**
 * Mode de cycle quotidien.
 */
enum class DailyMode(val label: String) {
    MONTHLY("Cycle mensuel"),
    WEEKLY("Jour de la semaine");
}
